src/phy/mod.rs

```diff
@@ -132,6 +137,33 @@ pub use self::tracer::Tracer;
 ))]
 pub use self::tuntap_interface::TunTapInterface;
 
+/// An ID that can be used to uniquely identify a packet to a [`Device`],
+/// sent or received by that same [`Device`]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq)]
+pub struct PacketId {
+    id: usize,
+}
+
+impl PacketId {
+    pub const fn none() -> Option<Self> {
+        None
+    }
+
+    pub fn id(&self) -> usize {
+        self.id
+    }
+
+    pub(crate) fn copy(&self) -> Self {
+        Self { id: self.id }
+    }
+
+    /// CAUTION: don't use too often...
```

Comment:
What is the reason of the `CAUTION`?

Reply:
From matrix: 

Perhaps a slightly too placeholder-y comment hahaha. Wanted to somehow indicate that obtaining a `PacketId` should generally not be done by calling `PacketId::new`, but rather from an `Interface` or `RawSocket`